core/main.go

```diff
@@ -72,59 +100,70 @@ func main() {
 				RootCAs:            caCertPool,
 				InsecureSkipVerify: *insecureSkipVerify,
 			},
+			TLSHandshakeTimeout: 5 * time.Second,
 		}
 	}
 
-	client := &http.Client{Timeout: 10 * time.Second, Transport: tr}
+	// 2/2 Build HTTP Client
+	client := &http.Client{Transport: tr, Timeout: 10 * time.Second}
 
-	a, err := api.NewClient(&api.Config{
-		Address:    secretServiceBaseURL,
-		HttpClient: client,
-	})
-	s := a.Sys()
-	inited := false
-	sealed := true
+	// Loop duration interval between Vault init and unseal retries
+	intervalDuration := time.Duration(*waitInterval) * time.Second
+	// Loop exit condition
+	loopExit := false
 
 	for {
-		inited, err = s.InitStatus()
-		if err != nil {
-			lc.Error(fmt.Sprintf("Error while checking the initialization status: %s", err.Error()))
-		} else {
-			k, err := initVault(s, config.SecretService.TokenPath, inited)
-			if err != nil {
-				lc.Error(fmt.Sprintf("Error while initializing the vault with info: %s", err.Error()))
+		sCode, _ := vaultHealthCheck(config, client)
+
+		switch sCode {
+		case 200:
+			lc.Info(fmt.Sprintf("Vault is initialized and unsealed (Status Code: %d).", sCode))
+			loopExit = true
+		case 429:
+			lc.Error(fmt.Sprintf("Vault is unsealed and in standby mode (Status Code: %d).", sCode))
+			loopExit = true
+		case 501:
+			lc.Info(fmt.Sprintf("Vault is not initialized (Status Code: %d). Starting initialisation and unseal phases.", sCode))
+			_, err = vaultInit(config, client, debug)
+			if err == nil {
+				_, err = vaultUnseal(config, client, debug)
+				if err == nil {
+					loopExit = true
+				}
 			}
-
-			sealed, err = unsealVault(s, k)
-			if err != nil {
-				lc.Error(err.Error())
+		case 503:
+			lc.Info(fmt.Sprintf("Vault is sealed (Status Code: %d). Starting unseal phase...", sCode))
+			_, err = vaultUnseal(config, client, debug)
+			if err == nil {
+				loopExit = true
 			}
-
-			if sealed == false {
-				lc.Info("Vault has been unsealed successfully.")
-				break
+		default:
+			if sCode == 0 {
+				lc.Error(fmt.Sprintf("Vault is in an unknown state. No Status code available."))
 			} else {
-				lc.Error("Vault is still under sealed status. Will retry again.")
+				lc.Error(fmt.Sprintf("Vault is in an unknown state. Status code: %d", sCode))
 			}
 		}
-		lc.Info(fmt.Sprintf("waiting %d seconds to retry checking the vault status.", *waitInterval))
-		time.Sleep(time.Second * time.Duration(*waitInterval))
-	}
 
-	_, _ = vaultHealthCheck(config, client)
+		if loopExit {
+			break
+		}
+		lc.Error(fmt.Sprintf("Next Vault Init/Unseal attempt in %d seconds.", *waitInterval))
```

Comment:
This should probably be at an info level rather than an error level

Reply:
@anonymouse64 Agree, this was probably there from the beginning, not in that batch of modifications.